api/types/network/network.go

```diff
@@ -40,32 +38,10 @@ type CreateRequest struct {
 	CheckDuplicate *bool `json:",omitempty"`
 }
 
-// Inspect is the body of the "get network" http response message.
-type Inspect struct {
-	Name       string                      // Name is the name of the network
-	ID         string                      `json:"Id"` // ID uniquely identifies a network on a single machine
-	Created    time.Time                   // Created is the time the network created
-	Scope      string                      // Scope describes the level at which the network exists (e.g. `swarm` for cluster-wide or `local` for machine level)
-	Driver     string                      // Driver is the Driver name used to create the network (e.g. `bridge`, `overlay`)
-	EnableIPv4 bool                        // EnableIPv4 represents whether IPv4 is enabled
-	EnableIPv6 bool                        // EnableIPv6 represents whether IPv6 is enabled
-	IPAM       IPAM                        // IPAM is the network's IP Address Management
-	Internal   bool                        // Internal represents if the network is used internal only
-	Attachable bool                        // Attachable represents if the global scope is manually attachable by regular containers from workers in swarm mode.
-	Ingress    bool                        // Ingress indicates the network is providing the routing-mesh for the swarm cluster.
-	ConfigFrom ConfigReference             // ConfigFrom specifies the source which will provide the configuration for this network.
-	ConfigOnly bool                        // ConfigOnly networks are place-holder networks for network configurations to be used by other networks. ConfigOnly networks cannot be used directly to run containers or services.
-	Containers map[string]EndpointResource // Containers contains endpoints belonging to the network
-	Options    map[string]string           // Options holds the network specific options to use for when creating the network
-	Labels     map[string]string           // Labels holds metadata specific to the network being created
-	Peers      []PeerInfo                  `json:",omitempty"` // List of peer nodes for an overlay network
-	Services   map[string]ServiceInfo      `json:",omitempty"`
-}
-
-// Summary is used as response when listing networks. It currently is an alias
-// for [Inspect], but may diverge in the future, as not all information may
-// be included when listing networks.
-type Summary = Inspect
+// Inspect is the body of the "get network" http response message. It currently
+// is an alias for [Summary], but may diverge in the future, as not all
+// information may be included when listing networks.
+type Inspect = Summary
```

Comment:
Do you think we should make them both concrete types? (and switching to the API module to be a good opportunity to do so?)

Reply:
Yeah, I think my intent with the alias was indeed to make them concrete types; sharing the types has been problematic in various cases where we (probably) were too lazy to define a new type, and just reused an existing type, leaving unused fields unset.

I'm not _entirely_ sure if we should embed the type (i.e. `Inspect` _must be_ a superset of `Summary` - it may be possible that fields could make sense for `Summary` that are a different presentation of fields in `Inspect`

Reply:
> it may be possible that fields could make sense for `Summary` that are a different presentation of fields in `Inspect`

Serving both presentations could save a round-trip for clients that need both for whatever reason?

I take the fact `Inspect` and `Summary` have been sharing a type definition all this time as a strong hint that there is a common set of properties that any API response describing a network should be including. What do you think of doing something more along the lines of this?
```go
type Summary struct {...}

type Inspect struct {
        Summary
        ...
}

type NetworkListResponseItem struct {
        Summary
        ...
}
```